UI/obs-app-theming.cpp

```diff
@@ -409,26 +409,29 @@ static vector<OBSThemeVariable> ParseThemeVariables(const char *themeData)
 
 void OBSApp::FindThemes()
 {
-	string themeDir;
 
 	QStringList filters;
 	filters << "*.obt" // OBS Base Theme
 		<< "*.ovt" // OBS Variant Theme
 		<< "*.oha" // OBS High-contrast Adjustment layer
 		;
 
-	GetDataFilePath("themes/", themeDir);
-	QDirIterator it(QString::fromStdString(themeDir), filters, QDir::Files);
-	while (it.hasNext()) {
-		auto theme = ParseThemeMeta(it.next());
-		if (theme && !themes.contains(theme->id))
-			themes[theme->id] = std::move(*theme);
+	{
+		string themeDir;
+		GetDataFilePath("themes/", themeDir);
+		QDirIterator it(QString::fromStdString(themeDir), filters,
+				QDir::Files);
+		while (it.hasNext()) {
+			auto theme = ParseThemeMeta(it.next());
+			if (theme && !themes.contains(theme->id))
+				themes[theme->id] = std::move(*theme);
+		}
 	}
 
-	themeDir.resize(1024);
-	if (GetConfigPath(themeDir.data(), themeDir.capacity(),
-			  "obs-studio/themes/") > 0) {
-		QDirIterator it(QT_UTF8(themeDir.c_str()), filters,
```

Comment:
IIRC, this was done this way to fix a specific bug with theme paths. We should confirm that the bug isn't reintroduced.

Reply:
@derrod I think you submitted the fix here with `.resize(1024)`. Do you remember what the test case was?

Reply:
That's just to ensure the buffer is big enough. Since the string is constructed directly here rather than being written by an external function to the underlying buffer that's fine.